dlt/extract/extract.py

```diff
@@ -48,12 +99,115 @@ def commit_extract_files(self, extract_id: str, with_delete: bool = True) -> Non
         if with_delete:
             self.storage.delete_folder(extract_path, recursively=True)
 
-    def _get_data_item_path_template(self, load_id: str, schema_name: str, table_name: str) -> str:
-        template = NormalizeStorage.build_extracted_file_stem(schema_name, table_name, "%s")
-        return self.storage.make_full_path(os.path.join(self._get_extract_path(load_id), template))
+    def write_data_item(self, file_format: TLoaderFileFormat, load_id: str, schema_name: str, table_name: str, item: TDataItems, columns: TTableSchemaColumns) -> None:
+        self.get_storage(file_format).write_data_item(load_id, schema_name, table_name, item, columns)
 
-    def _get_extract_path(self, extract_id: str) -> str:
-        return os.path.join(ExtractorStorage.EXTRACT_FOLDER, extract_id)
+
+def _item_format(item: TDataItems) -> TLoaderFileFormat:
```

Comment:
maybe a static method of Extract?

Reply:
also `item` may be a list here (`TDataItems`) so you need to check it for being a list first.

please also rename item to items. same for `_write_static_table`